include/morphio/collection.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <morphio/morphology.h>
+#include <morphio/mut/morphology.h>
+
+namespace morphio {
+
+class CollectionImpl
+{
+  public:
+    virtual ~CollectionImpl() = default;
+
+    virtual Morphology load(const std::string& morph_name) const = 0;
+    virtual mut::Morphology load_mut(const std::string& morph_name) const = 0;
+};
+
+class Collection
+{
+  public:
+    Collection(std::shared_ptr<CollectionImpl> collection);
+
+    /**
+     * Create a collection from the given path.
+     *
+     * If `collection_path` points to an HDF5 file, then that file
+     * must be a container. Otherwise the `collection_path` should
+     * point to the directory containing the morphology files.
+     *
+     * If the collection path is a directory, the extension of the morphology
+     * file must be guessed. The optional argument `extensions` specifies which
+     * and in which order the morphologies are searched.
+     */
+    Collection(std::string collection_path,
+               std::vector<std::string> extensions = std::vector<std::string>{"h5", "swc", "asc"});
```

Comment:
at bbp, I think we should put `swc` last; since it's generally not used.  Annoyingly, I'm not sure we're 100% standardized on lowercase `asc`, so adding `ASC` might be required :(  I will look around.